volmdlr/faces.py

```diff
@@ -642,40 +708,65 @@ def face_from_contours3d(self,
         else:
             class_ = self.face_class
 
-        surface2d = Surface2D(outer_contour=outer_contour2d,
-                              inner_contours=inner_contours2d)
-        return class_(self,
-                      surface2d=surface2d,
-                      name=name)
+        surface2d = Surface2D(outer_contour=outer_contour2d, inner_contours=inner_contours2d)
+        return class_(self, surface2d=surface2d, name=name)
+
+    # def repair_singularity(self, primitives, last_primitives):
 
     def repair_primitives_periodicity(self, primitives, last_primitive):
-        delta_x1 = abs(primitives[0].start.x
-                       - last_primitive.end.x)
-        delta_x2 = abs(primitives[-1].end.x
-                       - last_primitive.end.x)
-        delta_y1 = abs(primitives[0].start.y
-                       - last_primitive.end.y)
-        delta_y2 = abs(primitives[-1].end.y
-                       - last_primitive.end.y)
+        delta_x1 = abs(primitives[0].start.x - last_primitive.end.x)
+        delta_x2 = abs(primitives[-1].end.x - last_primitive.end.x)
+        delta_y1 = abs(primitives[0].start.y - last_primitive.end.y)
+        delta_y2 = abs(primitives[-1].end.y - last_primitive.end.y)
+
+        new_primitives = primitives
 
         if self.x_periodicity \
                 and not (math.isclose(delta_x1, 0,
                                       abs_tol=5e-5)
                          or math.isclose(delta_x2, 0,
                                          abs_tol=5e-5)):
+
             delta_x1 = delta_x1 % self.x_periodicity
             delta_x2 = delta_x2 % self.x_periodicity
+
             if math.isclose(delta_x1, self.x_periodicity,
                             abs_tol=1e-4):
                 delta_x1 = 0.
+
             if math.isclose(delta_x2, self.x_periodicity,
                             abs_tol=1e-4):
                 delta_x2 = 0.
+
+            delta = last_primitive.end - primitives[0].start
+            new_primitives = []
             for prim in primitives:
-                prim.start.x = abs(self.x_periodicity
-                                   - prim.start.x)
-                prim.end.x = abs(self.x_periodicity
-                                 - prim.end.x)
+                if isinstance(self, SphericalSurface3D):
```

Comment:
I hate this pattern if isinstance: can we switch to a method that each concept would implement